scripts/install_backend_python_libs.py

```diff
@@ -0,0 +1,276 @@
+              requirements_version > directory_version):
+            subprocess.check_call([
+                'pip', 'install', '--target',
+                common.THIRD_PARTY_PYTHON_LIBS_DIR,
+                '--no-dependencies',
+                '%s==%s' % (
+                    library,
+                    python_utils.convert_to_bytes(requirements_version)),
+                '--upgrade'
+            ])
+        # The currently installed library version is higher than the required
+        # 'requirements.txt' version.
+        elif requirements_version < directory_version:
+            subprocess.check_call([
+                'pip', 'install', '--target',
+                common.THIRD_PARTY_PYTHON_LIBS_DIR,
+                '--no-dependencies',
+                '%s==%s' % (
+                    library,
+                    python_utils.convert_to_bytes(requirements_version)),
+                '--upgrade'
+            ])
+            _remove_metadata(
+                library, python_utils.convert_to_bytes(directory_version))
+
+
+def get_mismatches():
+    """Returns a dictionary containing mismatches between the 'requirements.txt'
+    file and the 'third_party/python_libs' directory. Mismatches are defined as
+    the following inconsistencies:
+        1. A library exists in the requirements file but is not installed in the
+           'third_party/python_libs' directory.
+        2. A library is installed in the 'third_party/python_libs'
+           directory but it is not listed in the requirements file.
+        3. The library version installed is not as recent as the library version
+           listed in the requirements file.
+        4. The library version installed is more recent than the library version
+           listed in the requirements file.
+
+    Returns:
+        dict(string, tuple(string|None, string|None)). Dictionary with the
+        library names as keys and tuples as values. The 1st element of the
+        tuple is the version string of the library required by the
+        requirements.txt file while the 2nd element is the version string of
+        the library currently in the 'third_party/python_libs' directory. If
+        the library doesn't exist, the tuple element will be None.
+        For example, the following dictionary signifies that 'requirements.txt'
+        requires flask with version 1.0.1 while the 'third_party/python_libs'
+        directory contains flask 1.1.1 (or mismatch 4 above):
+            {
+              flask: ('1.0.1', '1.1.1')
+            }
+    """
+    requirements_contents = _get_requirements_file_contents()
+    directory_contents = _get_third_party_directory_contents()
+
+    mismatches = {}
+    for library in requirements_contents:
+        # Library exists in the directory and the requirements file.
+        if library in directory_contents:
+            # Library and version match.
+            if directory_contents[library] == requirements_contents[library]:
```

Comment:
You can simplify this -- change == to != and drop the next three lines. (And fix the comment above this line, too.)